fio/src/fiodriver/fioman.c

```diff
@@ -570,6 +570,43 @@ fioman_add_frame
         return PTR_ERR(rxframe);
 }
 
+/*****************************************************************************/
+/*
+This function is used to remove a specific request frame and response, if any
+*/
+/*****************************************************************************/
+int
+fioman_remove_frame
+(
+	FIOMAN_SYS_FIOD	*p_sys_fiod,		/* FIOD of destined frame */
+	int frame_no
+)
+{
+  struct list_head	*p_elem;	/* Ptr to queue element being examined */
+	struct list_head	*p_next;	/* Temp Ptr to next for loop */
+	FIOMSG_TX_FRAME		*p_tx_elem;	/* Ptr to tx frame being examined */
+	FIOMSG_PORT		*p_port;	/* Port of Request Queue */
+
+  /* Search for frame in tx queue */
+  /* Get port to work on */
+  p_port = FIOMSG_P_PORT(p_sys_fiod->fiod.port);
+  /* For each element in the queue */
+  list_for_each_safe(p_elem, p_next, &p_port->tx_queue) {
+    /* Get the request frame for this queue element */
+    p_tx_elem = list_entry(p_elem, FIOMSG_TX_FRAME, elem);
+    /* See if current element matches fiod */
+    if (p_tx_elem->fiod.fiod == p_sys_fiod->fiod.fiod) {
+      /* Does the frame number match one requested? */
+      if (FIOMSG_PAYLOAD(p_tx_elem)->frame_no == frame_no) {
+        list_del_init(p_elem);
+        kfree(p_tx_elem);
+      }
+    }
+  }
+  return(-EINVAL);
```

Comment:
Why return this value?

Reply:
Err, yes, that isn't right. I think it was meant to indicate not-found, but should be under an else condition. I'll fix that.